Core/src/RequestWrapper.php

```diff
@@ -282,38 +283,41 @@ private function applyHeaders(RequestInterface $request)
         ];
 
         if ($this->shouldSignRequest) {
-            $headers['Authorization'] = 'Bearer ' . $this->getToken();
-        }
+            $quotaProject = $this->quotaProject;
+            $token = null;
+
+            if ($this->accessToken) {
+                $token = $this->accessToken;
+            } else {
+                $credentialsFetcher = $this->getCredentialsFetcher();
+                $token = $this->fetchCredentials($credentialsFetcher);
+                $quotaProject = $credentialsFetcher->getQuotaProject() ?: $quotaProject;
```

Comment:
We need to update `google/auth` to ^1.9 in `composer.json`

Reply:
Another potential way to handle this is to surround the call in an interface check:

```php
if ($credentialsFetcher implements \Google\Auth\GetQuotaProjectInterface) {
    $quotaProject = $credentialsFetcher->getQuotaProject() ?: $quotaProject;
}
```

Reply:
I like option #2 :)